File: cpp/example/src/ContainsDuplicate2/containsDuplicate2.cpp
#include "containsDuplicate2.h"

bool containsDuplicate2::naive(vector<int>& nums,int k){
    map<int,int> prevIdxDict;
    map<int,int>::iterator it;
    for (int i=0;i<nums.size();i++){
        it = prevIdxDict.find(nums[i]);
        if (it!=prevIdxDict.end() && 
            abs(i-prevIdxDict[nums[i]])<=k) return true;
        prevIdxDict[nums[i]]=i;
    }
    return false;
}

File: cs/sample/Sample/ContainsDuplicate2/ContainsDuplicate2.cs
using System;
using System.Collections.Generic;

namespace Sample.ContainsDuplicate2{
    public class containsDuplicate2{
        public static bool naive(int[] nums,int k){
            Dictionary<int,int> prevIdxDict=new Dictionary<int,int>();
            for (int i=0;i<nums.Length;i++){
                if (prevIdxDict.ContainsKey(nums[i]) && 
                    Math.Abs(i-prevIdxDict[nums[i]])<=k) return true;
                prevIdxDict[nums[i]] = i;
            }
            return false;
        }
    }
}
